modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -392,11 +407,27 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
             throw new IllegalStateException("no document to percolate");
         }
 
-        MapperService mapperService = context.getMapperService();
-        DocumentMapperForType docMapperForType = mapperService.documentMapperWithAutoCreate(documentType);
-        DocumentMapper docMapper = docMapperForType.getDocumentMapper();
+        final ParsedDocument doc;
+        final DocumentMapper docMapper;
+        final MapperService mapperService = context.getMapperService();
+        if (context.getIndexSettings().isSingleType()) {
+            if (documentType != null) {
+                DEPRECATION_LOGGER.deprecated("[document_type] parameter has been deprecated because types have been deprecated");
+            }
 
-        ParsedDocument doc = docMapper.parse(source(context.index().getName(), documentType, "_temp_id", document, documentXContentType));
+            Collection<String> types = mapperService.types();
+            assert types.size() == 1;
```

Comment:
> it could be 0 if the index was just created without types?

That can happen, but that also mean that there isn't a percolator field in that case we should fail. I'll move the field validation before this part, so that check can remain the same.

> if types.size() == 1 should we throw an exception if documentType is different from type?

This  if statement only gets executed if `index.mapping.single_type` is set to true, so then there should be only a 1 type?